src/main/java/thePackmaster/actions/FlexibleDiscoveryAction.java

```diff
@@ -0,0 +1,59 @@
+package thePackmaster.actions;
+
+import basemod.BaseMod;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.unlock.UnlockTracker;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardAndAddToDiscardEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardAndAddToHandEffect;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class FlexibleDiscoveryAction extends AbstractGameAction {
```

Comment:
It's almost like I've seen this code elsewhere... copied into a dozen different mods.